easybuild/framework/easyblock.py

```diff
@@ -426,6 +429,30 @@ def fetch_patches(self, patch_specs=None, extension=False, checksums=None):
                 else:
                     raise EasyBuildError("Wrong patch spec '%s', only int/string are supported as 2nd element",
                                          str(patch_spec))
+
+            # patch_spec is dict
+            elif isinstance(patch_spec, collections.Mapping):
```

Comment:
Suggestion: As below a `patchspec` dict is created (mind the almost name clash), maybe don't do this checking here (I'd assume this is done by the parser already?) and simply coerce the tuple into a dict in the above `if` and also below. So basically: If it isn't already a dict, make it one, else assume it is valid. In any case add the remaining keys (name, path, checksum, ...)